source/common/config/config_provider_impl.cc

```diff
@@ -23,6 +23,20 @@ ConfigSubscriptionCommonBase::~ConfigSubscriptionCommonBase() {
   init_target_.ready();
   config_provider_manager_.unbindSubscription(manager_identifier_);
 }
+
+void ConfigSubscriptionCommonBase::applyConfigUpdate(const ConfigUpdateCb& update_fn,
+                                                     const Event::PostCb& complete_cb) {
+  tls_->runOnAllThreads(
+      [update_fn](ThreadLocal::ThreadLocalObjectSharedPtr previous)
```

Comment:
nit: `const ThreadLocal::ThreadLocalObjectSharedPtr&`

Reply:
I want to express the idea that: ThreadLocal::Slot::UpdateCb is expected to receive data stored in the slot, and do-whatever-necessary, then return the new/updated version data to be saved in the slot. A non-const ThreadLocalObjectSharedPtr seems better under this notion, WDYT?

Reply:
By passing by value you are essentially just forcing a copy in most cases. Personally, I think it's more common to make a new thing than use the previous, so I would pass by const-ref, and then if a copy needs to be made it can be done inside the function to put the cost there.